uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/absence-type-api.js

```diff
@@ -22,6 +23,29 @@ define([
           .then(function (data) {
             return data.values;
           });
+      },
+
+      /**
+       * Calculate Toil Expiry Date
+       *
+       * @param  {string} absenceTypeID
+       * @param  {Object} date
+       * @param  {Object} params
+       * @return {Promise}
+       */
+      calculateToilExpiryDate: function (absenceTypeID, date, params) {
+        $log.debug('calculateToilExpiryDate');
+
+        params = params ? params : {};
+        params = _.assign(params, {
+          absence_type_id: absenceTypeID,
+          date: date
+        });
+
+        return this.sendPOST('AbsenceType', 'calculateToilExpiryDate', params)
+          .then(function (data) {
+            return data.values.expiry_date;
```

Comment:
I think a basic check for isError will be useful even now, its been done in few other APIs